### PR TITLE
Feature/edit sprint3

### PR DESCRIPTION
**Pull request by tamaragovorcin**

### CODE REVIEW

File: IntegrationWithPharmacies/FileProtocol/ReportText.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                stringBuilder.Append(registration.ApiKey + ";");
+            }
+            return stringBuilder.ToString();
+        }
+
+        public String getReportText(DateOfOrder date)
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            int totalQuatity = 0;
+            int i = 1;
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/FileProtocol/ReportText.cs:42]: Remove unnecessary variable i

File: IntegrationWithPharmacies/FileProtocol/SmptServerService.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            SmtpClient SmptServer = new SmtpClient("smtp.gmail.com", 587);
+            SmptServer.Credentials = new System.Net.NetworkCredential("ourhospital9@gmail.com", "hospital.9");
+            SmptServer.EnableSsl = true;
+            SmptServer.Send(message);
+        }
+
+        private static MailMessage CreateMailMessage(String filePath, String type)
+        {
+            MailMessage message = GetMessageInformation(type);
+            Attachment data = new Attachment(filePath, MediaTypeNames.Application.Octet);
```
Comment (velicko-tam) [inline IntegrationWithPharmacies/FileProtocol/SmptServerService.cs:29]: Try removing data variable and adding it directly in line 30

File: PatientWebApplication/HealthClinic.DAL/Dtos/MedicineDescriptionDto.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public class MedicineDescriptionDto
+    {
+        public String Name { get; set; }
+        public String Description { get; set; }
+        public int MedicineDescriptionId { get; set; }
+
+        public MedicineDescriptionDto() { }
+
+        public MedicineDescriptionDto(string name, String description, int medicineID)
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Dtos/MedicineDescriptionDto.cs:15]: Change name of medicineID variable into medicineDescriptionId to match lines above

File: PatientWebApplication/HealthClinic.DAL/Dtos/MessageDto.cs [hunk 4]
```diff
@@ -8,19 +8,22 @@ public class MessageDto
...
         public DateTime TimeStamp { get; set; }
 
+        public string DateStamp { get; set; }
+
         public string PharmacyName { get; set; }
 
         public string DateAction { get; set; }
 
-        public MessageDto(string text, DateTime timestamp, string pharmacyname, string dateaction)
+        public MessageDto(string text, string datestamp, DateTime timestamp, string pharmacyname, string dateaction)
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Dtos/MessageDto.cs:17]: Chnage names of variables to match lines above, for example datestamp should be dateStamp, pharmacyname should be pharmacyName...

File: PatientWebApplication/HealthClinic.DAL/Model/ActionsAndBenefits/Message.cs [hunk 5]
```diff
@@ -10,32 +10,37 @@ public class Message : Entity
...
 
         public string DateAction { get; set; }
 
+
         public Message() : base()
         {
         }
 
-        public Message(int id, string text, DateTime timestamp, bool isremoved, string pharmacyname, string dateaction) : base(id)
+        public Message(int id, string text, string datestamp, DateTime timestamp, bool isremoved, string pharmacyname, string dateaction) : base(id)
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Model/ActionsAndBenefits/Message.cs:26]: Chnage names of variables to match lines above, for example datestamp should be dateStamp, pharmacyname should be pharmacyName...

File: PatientWebApplication/HealthClinic.DAL/Model/ActionsAndBenefits/Message.cs [hunk 6]
```diff
@@ -10,32 +10,37 @@ public class Message : Entity
...
             Text = text;
             TimeStamp = timestamp;
             IsRemoved = isremoved;
             PharmacyName = pharmacyname;
             DateAction = dateaction;
+            DateStamp = datestamp;
         }
 
-        public Message(string text, DateTime timestamp, bool isremoved, string pharmacyname, string dateaction)
+        public Message(string text, string datestamp, DateTime timestamp, bool isremoved, string pharmacyname, string dateaction)
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Model/ActionsAndBenefits/Message.cs:36]: Chnage names of variables to match lines above, for example datestamp should be dateStamp, pharmacyname should be pharmacyName...

File: PatientWebApplication/HealthClinic.DAL/Model/Orders/MedicineDescription.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public class MedicineDescription : Entity
+    {
+        public String Name { get; set; }
+        public String Description { get; set; }
+        public int MedicineDescriptionId { get; set; }
+
+        public MedicineDescription() : base() { }
+
+        public MedicineDescription(int id, string name, String description, int medicineID) : base(id)
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Model/Orders/MedicineDescription.cs:14]: Change variable name from medicineID to medicineDescriptionId to match lines above

File: PatientWebApplication/HealthClinic.DAL/Model/Orders/MedicineDescription.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        {
+            Name = name;
+            Description = description;
+            MedicineDescriptionId = medicineID;
+        }
+        public MedicineDescription(string name, String description, int medicineID)
```
Comment (velicko-tam) [inline PatientWebApplication/HealthClinic.DAL/Model/Orders/MedicineDescription.cs:20]: Change variable name from medicineID to medicineDescriptionId to match lines above

**Review by velicko-tam — CHANGES_REQUESTED** [review]
PR is not approved. Marked parts need to be fixed.

**Review by velicko-tam — APPROVED** [review]
PR is approved. Clean code principles were followed.

File: IntegrationWithPharmacies/FileProtocol/HttpService.cs [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            client.Credentials = CredentialCache.DefaultCredentials;
+            client.UploadFile(new Uri(@"http://localhost:8082/download/prescription/http"), "POST", complete);
+            client.Dispose();
+        }
+        public static String FormMedicineAvailabilityRequest(string medicine)
+        {
+            HttpWebRequest webRequest = (HttpWebRequest)WebRequest.Create("http://localhost:8082/medicinePharmacy/" + medicine);
+            HttpWebResponse webResponse = (HttpWebResponse)webRequest.GetResponse();
+            Stream response = webResponse.GetResponseStream();
+            return new StreamReader(response, System.Text.Encoding.GetEncoding("utf-8")).ReadToEnd();
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/HttpService.cs:31]: Write this in form:
"return new StreamReader(webResponse.GetResponseStream(), System.Text.Encoding.GetEncoding("utf-8")).ReadToEnd();"

File: IntegrationWithPharmacies/FileProtocol/HttpService.cs [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        public static string FormMedicineDescriptionRequest(string medicine)
+        {
+            HttpWebRequest webRequest = (HttpWebRequest)WebRequest.Create("http://localhost:8082/description/medicine/" + medicine);
+            HttpWebResponse webResponse = (HttpWebResponse)webRequest.GetResponse();
+            Stream response = webResponse.GetResponseStream();
+            return new StreamReader(response, System.Text.Encoding.GetEncoding("utf-8")).ReadToEnd();
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/HttpService.cs:38]: Write this in form:
"return new StreamReader(webResponse.GetResponseStream(), System.Text.Encoding.GetEncoding("utf-8")).ReadToEnd();"

File: IntegrationWithPharmacies/FileProtocol/HttpService.cs [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        public static IRestResponse<List<MedicineName>> FormMedicineFromIsaRequest()
+        {
+            var client = new RestSharp.RestClient("http://localhost:8082");
+            var response = client.Get<List<MedicineName>>(new RestRequest("/medicineRequested"));
+            response.Data.ForEach(medicine => Console.WriteLine(medicine.ToString()));
+            return response;
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/HttpService.cs:45]: Write this in form:
"return response.Data.ForEach(medicine => Console.WriteLine(medicine.ToString()));"

File: IntegrationWithPharmacies/FileProtocol/SmptServerService.cs [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+            catch (SmtpException exception) { Console.WriteLine(exception.Message); }
+        }
+
+        private static void SendMail(String filePath, String type)
+        {
+            MailMessage message = CreateMailMessage(filePath,type);
+            SmtpClient SmptServer = new SmtpClient("smtp.gmail.com", 587);
+            SmptServer.Credentials = new System.Net.NetworkCredential("ourhospital9@gmail.com", "hospital.9");
+            SmptServer.EnableSsl = true;
+            SmptServer.Send(message);
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/SmptServerService.cs:23]: Write this in form:
"SmptServer.Send(CreateMailMessage(filePath,type));"

File: IntegrationWithPharmacies/FileProtocol/SmptServerService.cs [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+        public void SendEMailNotification(String filePath, String type)
+        {
+            try { SendMail(filePath,type); }
+            catch (SmtpException exception) { Console.WriteLine(exception.Message); }
+        }
+
+        private static void SendMail(String filePath, String type)
+        {
+            MailMessage message = CreateMailMessage(filePath,type);
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/SmptServerService.cs:19]: Delete this object of MailMessage.

File: IntegrationWithPharmacies/FileProtocol/SmptServerService.cs [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+        private static MailMessage GetMessageInformation(string type)
+        {
+            if (type.Equals("report"))
+            {
+                return new MailMessage("ourhospital9@gmail.com", "pharmacyisa@gmail.com", "Notification about new report about medicine consumption", "Body of mail address");
+            }
+            else
+            {
+                return new MailMessage("ourhospital9@gmail.com", "pharmacyisa@gmail.com", "Notification about new prescription", "Body of mail address");
+            }
```
Comment (trki293) [inline IntegrationWithPharmacies/FileProtocol/SmptServerService.cs:42]: Write the selected code in the form of a ternary operator.

File: IntegrationWithPharmacies/Services/ClientScheduledService.cs [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        private Channel channel;
+        private SpringGrpcService.SpringGrpcServiceClient client;
+
+        public ClientScheduledService() { }
+
+        public async Task<string> SendMessage(string name)
+        {
+            channel = new Channel("127.0.0.1:8787", ChannelCredentials.Insecure);
+            client = new SpringGrpcService.SpringGrpcServiceClient(channel);
```
Comment (trki293) [inline IntegrationWithPharmacies/Services/ClientScheduledService.cs:17]: Write this in form:
"client = new SpringGrpcService.SpringGrpcServiceClient(new Channel("127.0.0.1:8787", ChannelCredentials.Insecure));"

File: IntegrationWithPharmacies/Services/RabbitMQService.cs [hunk 16]
```diff
@@ -1,58 +1,62 @@
...
-                Message message;
-                try{  message = JsonConvert.DeserializeObject<Message>(jsonMessage); }
-                catch (Exception)   
-                {
-                    message = JsonConvert.DeserializeObject<Message>(jsonMessage, new MyDateTimeConverter());
-                }
-                Program.ListOfMessages.Add(message);
-             
-                MessageService messageService = new MessageService();
-                //messageService.Create(new MessageDto(message.Text, message.TimeStamp, "Apoteka Jankovic", message.DateAction));
```
Comment (trki293) [inline IntegrationWithPharmacies/Services/RabbitMQService.cs:39]: Delete commented line of code.

File: PatientWebApplication/HealthClinic.DAL/Dtos/MessageDto.cs [hunk 17]
```diff
@@ -8,19 +8,22 @@ public class MessageDto
...
         public DateTime TimeStamp { get; set; }
 
+        public string DateStamp { get; set; }
+
         public string PharmacyName { get; set; }
 
         public string DateAction { get; set; }
 
-        public MessageDto(string text, DateTime timestamp, string pharmacyname, string dateaction)
+        public MessageDto(string text, string dateStamp, DateTime timeStamp, string pharmacyName, string dateaction)
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Dtos/MessageDto.cs:17]: Rename parameter 'dateaction' to 'dateAction'.

File: PatientWebApplication/HealthClinic.DAL/Model/ActionsAndBenefits/Message.cs [hunk 18]
```diff
@@ -10,32 +10,35 @@ public class Message : Entity
...
         public string DateAction { get; set; }
 
-        public Message() : base()
-        {
-        }
 
-        public Message(int id, string text, DateTime timestamp, bool isremoved, string pharmacyname, string dateaction) : base(id)
+        public Message() : base(){ }
+
+        public Message(int id, string text, string dateStamp, DateTime timeStamp, bool isRemoved, string pharmacyName, string dateaction) : base(id)
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Model/ActionsAndBenefits/Message.cs:24]: Rename parameter 'dateaction' to 'dateAction'.

File: PatientWebApplication/HealthClinic.DAL/Model/ActionsAndBenefits/Message.cs [hunk 19]
```diff
@@ -10,32 +10,35 @@ public class Message : Entity
...
-            PharmacyName = pharmacyname;
+            TimeStamp = timeStamp;
+            IsRemoved = isRemoved;
+            PharmacyName = pharmacyName;
             DateAction = dateaction;
+            DateStamp = dateStamp;
         }
 
-        public Message(string text, DateTime timestamp, bool isremoved, string pharmacyname, string dateaction)
+        public Message(string text, string dateStamp, DateTime timeStamp, bool isRemoved, string pharmacyName, string dateaction)
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Model/ActionsAndBenefits/Message.cs:34]: Rename parameter 'dateaction' to 'dateAction'.

File: PatientWebApplication/HealthClinic.DAL/Service/EPrescriptionService.cs [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            List<EPrescription> prescriptions = GetAllForStub();
+            foreach (EPrescription prescription in prescriptions)
+            {
+                if (prescription.MedicalIDNumber.Equals(medicalIdNumber))
+                {
+                    return getTextForPrescription(prescription);
+                }
+            }
+            return "";
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/EPrescriptionService.cs:50]: Write this in form:
"EPrescription prescription = GetAllForStub().SingleOrDefault(prescriptionIt => (prescriptionIt.MedicalIDNumber.Equals(medicalIdNumber)));
return (prescritption!=null?getTextForPrescription(prescription):"");"

File: PatientWebApplication/HealthClinic.DAL/Service/MedicineDescriptionService.cs [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+        public string GetMedicineDescriptionFromStub(string medicineName)
+        {
+            foreach (MedicineDescription medicine in GetAllForStub())
+            {
+                if (medicine.Name.Equals(medicineName))
+                {
+                    return medicine.Description;
+                }
+            }
+            return "";
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/MedicineDescriptionService.cs:49]: Write this in form:
"MedicineDescription medicineDescription= GetAllForStub().SingleOrDefault(medicineDescriptionIt => (medicineDescriptionIt.Name.Equals(medicineName)));
return (medicineDescription!=null?medicine.Description:"");"

File: PatientWebApplication/HealthClinic.DAL/Service/MedicineDescriptionService.cs [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+        public String GetMedicineDescriptionFromDatabase(String medicine)
+        {
+            foreach (MedicineDescription medicineDescription in GetAll())
+            {
+                if (medicineDescription.Name.ToString().Equals(medicine))
+                {
+                    return medicineDescription.Description.ToString();
+                }
+            }
+            return "";
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/MedicineDescriptionService.cs:60]: Write this in form:
"MedicineDescription medicineDescription= GetAll().SingleOrDefault(medicineDescriptionIt => (medicineDescriptionIt.Name.Equals(medicine)));
return (medicineDescription!=null?medicine.Description:"");"

File: PatientWebApplication/HealthClinic.DAL/Service/MedicineDescriptionService.cs [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+            foreach (MedicineDescription medicine in GetAllForStub())
+            {
+                if (medicine.Name.Equals(medicineName))
+                {
+                    return medicine.Description;
+                }
+            }
+            return "";
+        }
+        public String GetMedicineDescriptionFromDatabase(String medicine)
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/MedicineDescriptionService.cs:51]: Rename parameter 'medicine' to 'medicineName' and change to return the method value to a string.

File: PatientWebApplication/HealthClinic.DAL/Service/MedicineDescriptionService.cs [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+            return "";
+        }
+
+        public MedicineDescription createIMedicineDescription(MedicineDescriptionDto dto)
+        {
+            foreach (MedicineDescription medicine in IMedicineDescriptionRepository.GetAll())
+            {
+                if (medicine.Name.Equals(MedicineDescriptionAdapter.MedicineDescriptionDtoToMedicineDescription(dto).Name)) return null;
+            }
+            return MedicineDescriptionAdapter.MedicineDescriptionDtoToMedicineDescription(dto);
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/MedicineDescriptionService.cs:69]: Write the body of this method using the SingleOrDefault () method and the ternary operator for the return value.

**Review by trki293 — CHANGES_REQUESTED** [review]
Necessary changes in the code, in order to meet the principles of the Clean code.

**Review by trki293 — APPROVED** [review]
PR is approved. Good Clean code practice has been followed.

**Pull request merged**